src/lib.rs

```diff
@@ -1394,8 +1403,12 @@ impl<T: Counter> Histogram<T> {
         let mut min_i = None;
         let mut total_count: u64 = 0;
         for i in 0..until {
+            // TODO can panic. May not be a sensible place to use Result, but should audit paths
+            // that can get here.
             let count = self[i];
             if count != T::zero() {
+                // TODO don't unwrap here; weird types may not work.
+                // Fix Counter types to just be u8-64?
```

Comment:
The only thing I can think of is using `BigInt`, but I suspect that would run into other issues entirely. Limiting to `u8-u64` seems fine, but I'm not sure how we do that nicely (except I guess we could add our own `Num`-like trait and `impl` only for the integer types we want).